### PR TITLE
Reuse authinfo in context handler

### PR DESCRIPTION
**Pull request by HenningNormann**
Avoid redundant auth info calls for different actions within the same resource

### CODE REVIEW

File: src/Authorization/Services/Implementation/ContextHandler.cs [hunk 1]
```diff
@@ -103,7 +99,13 @@ protected async Task EnrichResourceAttributes(XacmlContextRequest request, bool
                 else
                 {
                     instanceData = new();
-                    AuthInfo authInfo = await _policyInformationRepository.GetAuthInfo(resourceAttributes.InstanceValue);
+                    
+                    if (!authInfoCache.TryGetValue(resourceAttributes.InstanceValue, out AuthInfo authInfo))
```
Comment (Alxandr) [inline src/Authorization/Services/Implementation/ContextHandler.cs:103]: This is most likely not thread safe

Comment (HenningNormann) [reply src/Authorization/Services/Implementation/ContextHandler.cs:103]: The cache is an instance variable from the controller. It's only used for evaluations for one decision request. The processing within a decision request is always sequential.

File: src/Authorization/Controllers/DecisionController.cs [hunk 2]
```diff
@@ -52,6 +53,8 @@ public class DecisionController : ControllerBase
         private readonly IAccessManagementWrapper _accessManagement;
         private readonly IMapper _mapper;
 
+        private readonly SortedDictionary<string, AuthInfo> _authInfoCache = new();
```
Comment (jonkjetiloye) [inline src/Authorization/Controllers/DecisionController.cs:56]: Can you please rename this instanceInfoCache or something so it actually describes what it is.
The AuthInfo model naming is way too vague of a naming for something used in Authorization context.

**Review by jonkjetiloye — CHANGES_REQUESTED** [review]

File: src/Authorization/Controllers/DecisionController.cs (hunk 2 above)
Comment (HenningNormann) [reply src/Authorization/Controllers/DecisionController.cs]: Have renamed authInfoCache to authInfoForOneDecReq

Comment (jonkjetiloye) [reply src/Authorization/Controllers/DecisionController.cs]: wow, that's even worse 😆
My issue is with AuthInfo. It doesn't contain authorization information. It's AppInstanceInfo.
That it's a local cache per. decision request is less relevant, or could be a comment.

**Review by jonkjetiloye — APPROVED** [review]

**Pull request merged**